packages/types/src/interfaces/extrinsic/v1/Extrinsic.ts

```diff
@@ -0,0 +1,110 @@
+import { GenericExtrinsic } from '@polkadot/types';
+import { SignatureOptions } from '@cennznet/types/interfaces/extrinsic/types';
+import { AnyNumber, AnyU8a, ExtrinsicPayloadValue, IKeyringPair, Registry } from '@polkadot/types/types';
+import { Address, Call } from '@polkadot/types/interfaces/runtime';
+import { ExtrinsicValueV4 } from '@polkadot/types/extrinsic/v4/Extrinsic';
+import { Api } from '@cennznet/api';
+import { EstimateFeeParams } from '@cennznet/api/derives/types';
+import { AnyJson, BareOpts } from '@polkadot/types/types/codec';
+
+export default class CENNZnetExtrinsic extends GenericExtrinsic{
+  private signaturePayloadOptions: SignatureOptions | ExtrinsicPayloadValue;
+
+  constructor(registry: Registry, value?: GenericExtrinsic | ExtrinsicValueV4 | AnyU8a | Call) {
+    super(registry, value);
+    this.signaturePayloadOptions = {
+      blockHash: undefined,
+      era: undefined,
+      genesisHash: undefined,
+      method: undefined,
+      specVersion: undefined,
+      nonce: undefined,
+      runtimeVersion: undefined,
+      signedExtensions: undefined,
+      signer: undefined,
+      tip: undefined,
+      transactionPayment: undefined,
+      transactionVersion: undefined
+    };
+  }
+
+  setNonce(value: AnyNumber){
+    this.signaturePayloadOptions.nonce = value;
+    return this;
+  }
+
+  addSignature(signer: Address | Uint8Array | string, signature: Uint8Array | string, payload: ExtrinsicPayloadValue | Uint8Array | string): GenericExtrinsic {
+    const mergeDefinedObjects = (A, B) => {
+      const res = {};
+      Object.keys({...A,...B}).map(key => {
+        res[key] = A[key] || B[key];
+      });
+      return res as ExtrinsicPayloadValue;
+    };
+
+    const mergedPayload = mergeDefinedObjects(payload, this.signaturePayloadOptions);
+    return super.addSignature(signer, signature, mergedPayload);
+  }
+
+  /**
+   * @description Sign the extrinsic with a specific keypair
+   */
+  sign(account: IKeyringPair, options: SignatureOptions): GenericExtrinsic {
+    const mergeDefinedObjects = (A, B) => {
+      const res = {};
+      Object.keys({...A,...B}).map(key => {
+          res[key] = A[key] || B[key];
+      });
+      return res as SignatureOptions;
+    };
+    const mergedSignatureOpts = mergeDefinedObjects(options, this.signaturePayloadOptions);
+    return super.sign(account, mergedSignatureOpts);
+  }
+
+  /**
+   * @description Adds payment options such as slippage and tip to the Extrinsic signature
+   */
+  paymentOpts(api: Api, feeAssetId: number | string, slippage = 0, tip = 0): this{
```

Comment:
let's call it `setPaymentOpts` for consistency
also providing de-structuring will help for readability

```suggestion
  setPaymentOpts(api: Api, { feeAssetId: number | string, slippage = 0, tip = 0 }): this{
```

Reply:
yup sure can change